corehq/apps/hqwebapp/session_details_endpoint/views.py

```diff
@@ -52,7 +53,13 @@ def post(self, request, *args, **kwargs):
 
         # reset the session's expiry if there's some formplayer activity
         secure_session = session.get('secure_session')
-        timeout = settings.SECURE_TIMEOUT if secure_session else settings.INACTIVITY_TIMEOUT
+        secure_session_timeout = settings.SECURE_TIMEOUT
+        domain = data.get('domain')
+        if domain:
+            domain_obj = Domain.get_by_name(domain)
+            if domain_obj:
+                secure_session_timeout = domain_obj.secure_timeout or secure_session_timeout
+        timeout = secure_session_timeout if secure_session else settings.INACTIVITY_TIMEOUT
```

Comment:
Could avoid domain lookup and all secure session logic entirely if `secure_session` is false.
```suggestion
        if secure_session:
            timeout = settings.SECURE_TIMEOUT
            domain = data.get('domain')
            if domain:
                domain_obj = Domain.get_by_name(domain)
                if domain_obj and domain_obj.secure_timeout:
                    timeout = domain_obj.secure_timeout
        else:
            timeout = settings.INACTIVITY_TIMEOUT
```